### PR TITLE
QGIS: fix color ramp

### LINKED ISSUE #230: review different color tables applied in QGIS
image

### PR DESCRIPTION
**Pull request by landam**
Fix #230

### CODE REVIEW

**Comment by landam** [comment]
Issue to be solved: wrong legend

image

Open symbology dialog, click OK:

image

**Comment by landam** [comment]
See https://gis.stackexchange.com/questions/420407/pyqgis-legend-doesnt-update-with-renderer

**Comment by landam** [comment]
Tested on Linux (Debian) and MS Windows. Seems to be fixed.

VirtualBox_MS Windows_13_10_2023_11_23_49

File: smoderp2d/__init__.py [hunk 1]
```diff
@@ -224,35 +224,38 @@ def import_data(options):
...
-        color_ramp.setColorRampType(QgsColorRampShader.Interpolated)
-        lst2 = [
-            QgsColorRampShader.ColorRampItem(0, QColor('#0000ff'), '0'),
-            QgsColorRampShader.ColorRampItem(20, QColor('#efefff'), '20')
-        ]
-        color_ramp.setColorRampItemList(lst2)
-        shader = QgsRasterShader()
-        shader.setRasterShaderFunction(color_ramp)
+        renderer = QgsSingleBandPseudoColorRenderer(layer.dataProvider(), 1)
+        color_ramp = QgsGradientColorRamp(QColor(239, 239, 255), QColor(  0,   0, 255))
```
Comment (pesekon2) [inline smoderp2d/__init__.py:238]:
```suggestion
        color_ramp = QgsGradientColorRamp(QColor(239, 239, 255), QColor(0,   0, 255))
```

File: smoderp2d/__init__.py [hunk 2]
```diff
@@ -224,35 +224,38 @@ def import_data(options):
...
+        renderer.setClassificationMax(stats.maximumValue)
+        renderer.createShader(color_ramp)
+
+        return renderer
+
+    def show_results(self):
+        import glob
+        from qgis.core import (
+            QgsProject, QgsRasterLayer, QgsSingleBandPseudoColorRenderer
+        )
```
Comment (pesekon2) [inline smoderp2d/__init__.py:249]:
```suggestion
        from qgis.core import QgsProject, QgsRasterLayer
```

**Review by pesekon2 — CHANGES_REQUESTED** [review]

**Comment by pesekon2** [comment]
Works also at my machine:

image

@landam Please, see two small change requests above and merge.

**Review by pesekon2 — APPROVED** [review]

**Pull request merged**